thinc/layers/premap_ids.pyx

```diff
@@ -0,0 +1,85 @@
+# cython: infer_types=True
+cimport cython
+import numpy as np
+from preshed.maps cimport PreshMap
+from preshed.maps import PreshMap
+from typing import Dict, Union, Optional, cast, Callable, Tuple, Mapping
+from ..types import Ints1d, Ints2d
+from ..config import registry
+from ..model import Model
+from ..util import to_numpy
+
+
+InT = Union[Ints1d, Ints2d]
+OutT = Ints2d
+
+
+cdef lookup(PreshMap mapping, long[:] keys, long default):
+    """
+    Faster dict.get(keys, default) for the case when 
+    the "dict" is a Dict[int, int] converted to PreshMap
+    and the "keys" is a numpy integer vector.
+    """
+    cdef int maxi = len(keys)
+    result = np.empty(maxi, dtype="int")
+    cdef long[:] result_view = result
+    for i in range(maxi):
+        v = mapping[keys[i]]
+        if v is None:
+            result_view[i] = default
+        else:
+            result_view[i] = v
+    return result
+
+
+@cython.binding(True)
```

Comment:
Elsewhere we usually add `binding=True` in the header line rather than using individual decorators.

Reply:
good idea, i just hate these headers because they can silently fail without syntax error or anything.